weights.py

```diff
@@ -189,19 +220,45 @@ def download_safetensors(url: str, path: Path):
         raise RuntimeError(f"Failed to download safetensors file: {e}")
 
 
-def make_download_url(url: str) -> str:
+def make_download_url(
+    url: str, hf_api_token: str | None = None, civitai_api_token: str | None = None
+) -> str:
     if url.startswith("data:"):
         return url
-    if m := re.match(r"^(?:https?://)?huggingface\.co/([^/]+)/([^/]+)/?$", url):
-        owner, model_name = m.groups()
-        return make_huggingface_download_url(owner, model_name)
+    if m := re.match(
+        r"^(?:https?://)?huggingface\.co/([^/]+)/([^/]+)(?:/([^/]+\.safetensors))?/?$",
+        url,
+    ):
+        if len(m.groups()) != 3 or not m.groups()[2]:
+            raise ValueError(
+                "Invalid HuggingFace URL. Expected format: huggingface.co/<owner>/<model-name>/<lora-weights-file.safetensors>"
+            )
+
+        owner, model_name, lora_weights = m.groups()
+        # Use HuggingFace Hub download directly
+        try:
+            token_value = (
+                hf_api_token.get_secret_value() if hf_api_token is not None else None
+            )
+            safetensors_path = hf_hub_download(
+                repo_id=f"{owner}/{model_name}",
+                filename=lora_weights,
+                token=token_value,
+            )
+            # Return the local path as a data URL to bypass normal download
+            print(f"Downloaded {lora_weights} from HuggingFace to {safetensors_path}")
+            return f"file://{safetensors_path}"
+        except Exception as e:
+            raise ValueError(f"Failed to download from HuggingFace: {e}")
     if m := re.match(r"^(?:https?://)?civitai\.com/models/(\d+)(?:/[^/?]+)?/?$", url):
         model_id = m.groups()[0]
-        return make_civitai_download_url(model_id)
+        return make_civitai_download_url(model_id, civitai_api_token)
     if m := re.match(r"^((?:https?://)?civitai\.com/api/download/models/.*)$", url):
         return url
```

Comment:
I believe these are redundant regexes - the first regex only matches up to `.safetensors` but returns the entire url; the second regex matches up to and after `.safetensors` and returns the entire url. 

since the new plan is to just return the whole url as long as `.safetensors` is in there somewhere, I don't think we need both.